### PR TITLE
Audit sweep #2: address #222-#228

### PR DESCRIPTION
**Pull request by hyiger**
## Summary

Addresses all seven issues from the second audit pass (filed against v1.16.1). Each commit maps 1:1 to an issue so review can proceed issue-by-issue.

- **#222 (P1 security)** — `PUT /api/filaments/{id}` and `POST /api/filaments` accepted `_purged: true` in the request body. The flag persisted on the doc, and the next hybrid-sync cycle would propagate the destructive tombstone to the peer DB — bypassing the trash → permanent-delete UI gate entirely. Added a one-line strip in both handlers plus the import-atlas destructure, with `tests/filament-purged-strip.test.ts` (3 cases) locking it in. The bypass was reproduced end-to-end during the audit before the fix landed.
- **#226 (P1 docs)** — `public/openapi.json` info.version was stuck at 1.13.0 (current 1.17.0). CLAUDE.md was missing v1.14 / v1.15 / v1.16 / OrcaSlicer-integration sections and had stale counts (868+ tests → ~1100, 737+ i18n keys → ~1100, "11k+ materials" → reference live `totalFFF`). Refreshed.
- **#223 (P2)** — three routes resolved variant-inherited fields by reading the variant doc directly: `/spool-check` (spoolWeight + density + diameter), `/analytics` (cost in both PrintHistory and per-spool-manual loops), and `/restore` (refused-orphan check missing). Each loads the parent inline using the same shape as PR #190's compare-route fix. `tests/variant-inheritance-routes.test.ts` (5 cases).
- **#224 (P2)** — print-history POST had no OCC between jobs and the standalone-fallback had no rollback on partial-write failure. Enabled `optimisticConcurrency: true` on the Filament schema, route now surfaces VersionError as 409, and the fallback now snapshots spool weights pre-mutation so it can reload + restore on save failure. `tests/print-history-concurrency.test.ts` (5 cases).
- **#225 (P2)** — `/api/openprinttag` first-cold-fetch occasionally 500s with TimeoutError. Wrapped the fetch+extract pipeline in a 3-attempt retry with exponential backoff; serves the stale cache (even past 1h TTL) if all retries fail. Added `Content-Security-Policy` header alongside the existing X-Content-Type-Options / X-Frame-Options / Referrer-Policy trio. Inline rationale spells out the `'unsafe-inline'` carve-out reasoning and the tightening path.
- **#227 (P2)** — coverage gap fills: the transactional `withTransaction` callback (previously only exercised via the standalone-fallback in tests), `tdsExtractor` HTTP 401 / HTTP 5xx / unknown-provider error branches, and the route-level translation of `CsvRowLimitExceededError` on `/api/spools/import`. `tests/audit-coverage-gaps.test.ts` (6 cases).
- **#228 (P3)** — polish bucket: CSV resurrect path now runs validators, refund clamps at `netFilamentWeight` ceiling, OpenAPI documents `?raw=true` on GET filament-by-id, `isPrivateIp` now blocks `192.0.0.0/24` + `198.18.0.0/15` (with test updates), and `expandAndScrollToSection` is finally wired into the form via `onInvalidCapture` so a required field in a collapsed section pops the section open before the browser tries to scroll to the invisible input.

## Test plan

- [x] `npm run lint` — clean
- [x] `npx tsc --noEmit` — clean
- [x] `npm test` — **1132 / 1132 passing** (was 1106; +26 new across 5 new test files)
- [x] `npm run build` — succeeds in standalone mode
- [ ] CI matrix (Node 20, 22) + smoke job pass on this PR
- [ ] Manual: PUT `{ "_purged": true }` against an active filament → field is not persisted (regression check for #222)
- [ ] Manual: open `/openprinttag` immediately after server boot, confirm no 500 even on a slow first connection (#225)
- [ ] Manual: open `/api-docs` and confirm CSP header is set on the response

Closes #222, closes #223, closes #224, closes #225, closes #226, closes #227, closes #228.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by hyiger** [comment]
@chatgpt-codex-connector please review

**Comment by hyiger** [comment]
@codex review

File: src/app/api/print-history/[id]/route.ts [hunk 1]
```diff
@@ -35,9 +35,21 @@ export async function DELETE(
...
+      // upper bound). The filament-level cap is correct because every
+      // spool of the same filament shares the same full-spool weight.
       if (typeof spool.totalWeight === "number") {
-        spool.totalWeight = spool.totalWeight + u.grams;
+        const refunded = spool.totalWeight + u.grams;
+        const capacity = filament.netFilamentWeight;
+        spool.totalWeight =
+          typeof capacity === "number" && capacity > 0
+            ? Math.min(refunded, capacity)
+            : refunded;
```
[the remark this answers is not in the document]
Comment (hyiger) [reply src/app/api/print-history/[id]/route.ts]: Fixed in 6e13b594. The gross ceiling now uses `spoolWeight + netFilamentWeight` and resolves both from the parent on variants (mirroring the GH #223 fixes for the same inheritance pattern). Four new test cases in `tests/print-history.test.ts` lock in the gross-clamp behaviour, the over-cap edge case, variant tare inheritance, and the legacy no-clamp path for filaments with no `netFilamentWeight` set.
